src/core/Target.ml

```diff
@@ -27,10 +27,13 @@ type lockfile = {
 }
 [@@deriving show]
 
+type scan_mode = All | InterOnly [@@deriving show]
+
 type regular = {
   path : path;
   analyzer : Xlang.t;
   products : Semgrep_output_v1_t.product list;
+  mode : scan_mode;
```

Comment:
I checked the Pro PR. I am not sure we need to change this core data type with something that is so specific to inter-file diff scans, when we could just return a pair of `Target.regular * scan_mode`.

Reply:
Although I'm not even sure about the solution itself, see comments in the Pro PR.